### PR TITLE
[ISSUE #1344] Fix /dev/shm not found on some OSs

### PR DESCRIPTION
**Pull request by liuruiyiyang**
## What is the purpose of the change

Fix issue #1344 

## Brief changelog

1. If the current OS is Darwin, create ```RAMDisk``` as the substitute for ```/dev/shm``` to store gc log. On other systems, check if  ```/dev/shm``` exists. If ```/dev/shm``` does not exist, store gc log in ```${BASE_DIR}```.

2. Correct and unify the gc log file name.

3. Use ```UseGCLogFileRotation``` in ```runserver.sh``` because name service is also a long-lived service process

modified:   distribution/benchmark/runclass.sh
modified:   distribution/bin/runbroker.sh
modified:   distribution/bin/runserver.sh

## Verifying this change

Start the name service and broker on the MacOS and Linux Systems

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by liuruiyiyang** [comment]
@vongosling  @duhenglucky PTAL.

File: distribution/benchmark/runclass.sh [hunk 1]
```diff
@@ -24,13 +24,39 @@ fi
 BASE_DIR=$(dirname $0)/..
 CLASSPATH=.:${BASE_DIR}/conf:${CLASSPATH}
 
+choose_gc_log_directory()
+{
+    case "`uname`" in
+        Darwin)
+            if [ ! -d "/Volumes/RAMDisk" ]; then
+                # create ram disk on Darwin systems as gc-log directory, default size is 600MB
+                DIR_SIZE_IN_MB=600
```
Comment (vongosling) [inline distribution/benchmark/runclass.sh]: This size could be a variable.

Comment (liuruiyiyang) [reply distribution/benchmark/runclass.sh]: Do you mean that I should put it in a config file? or move it outside the choose_gc_log_directory function as a parameter?

Comment (vongosling) [reply distribution/benchmark/runclass.sh]: The latter, the first step.

Comment (liuruiyiyang) [reply distribution/benchmark/runclass.sh]: OK, modified.

**Pull request merged**